### PR TITLE
Add support for query row

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
$subject

Related issue: ballerina-platform/ballerina-standard-library#1604

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

**Review by niveathika — DISMISSED** [review]

File: changelog.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 - [Change default rowType of the query remote method from `nil` to `<>`](https://github.com/ballerina-platform/ballerina-standard-library/issues/1445)
+- [Add support for queryRow](https://github.com/ballerina-platform/ballerina-standard-library/issues/1604)
```
Comment (lahirusamith) [inline changelog.md:16]:
```suggestion

### Added
- [Add support for queryRow](https://github.com/ballerina-platform/ballerina-standard-library/issues/1604)
```

File: ballerina/tests/simple-params-query-test.bal [hunk 2]
```diff
@@ -673,6 +673,110 @@ function queryJsonParam3() returns record {}|error? {
     validateJsonTable(queryMysqlClient(sqlQuery, resultType = JsonResult));
 }
 
+@test:Config {
+    groups: ["queryRow","query-simple-params"]
```
Comment (lahirusamith) [inline ballerina/tests/simple-params-query-test.bal:677]:
```suggestion
    groups: ["query", "query-simple-params"]
```

File: ballerina/tests/simple-params-query-test.bal [hunk 3]
```diff
@@ -673,6 +673,110 @@ function queryJsonParam3() returns record {}|error? {
...
+function queryRecord() returns sql:Error? {
+    int rowId = 1;
+    sql:ParameterizedQuery sqlQuery = `SELECT * from DataTable WHERE row_id = ${rowId}`;
+    Client dbClient = check new (host, user, password, simpleParamsDb, port);
+    record {} queryResult = check dbClient->queryRow(sqlQuery);
+    validateDataTableResult(queryResult);
+}
+
+@test:Config {
+    groups: ["queryRow","query-simple-params"]
```
Comment (lahirusamith) [inline ballerina/tests/simple-params-query-test.bal:688]:
```suggestion
    groups: ["query", "query-simple-params"]
```

File: ballerina/tests/simple-params-query-test.bal [hunk 4]
```diff
@@ -673,6 +673,110 @@ function queryJsonParam3() returns record {}|error? {
...
+    if queryResult is sql:Error {
+        test:assertTrue(queryResult is sql:NoRowsError);
+        test:assertTrue(queryResult.message().endsWith("Query did not retrieve any rows."), "Incorrect error message");
+   } else {
+       test:assertFail("Expected no rows error with empty query result.");
+   }
+}
+
+@test:Config {
+    groups: ["queryRow", "query-simple-params"]
```
Comment (lahirusamith) [inline ballerina/tests/simple-params-query-test.bal:704]:
```suggestion
    groups: ["query", "query-simple-params"]
```

File: ballerina/tests/simple-params-query-test.bal [hunk 5]
```diff
@@ -673,6 +673,110 @@ function queryJsonParam3() returns record {}|error? {
...
+    record{}|int|error queryResult = dbClient->queryRow(sqlQuery);
+    if queryResult is error {
+        test:assertEquals(queryResult.message(), "Return type cannot be a union.");
+    } else {
+        test:assertFail("Expected error when querying with union return type.");
+    }
+}
+
+@test:Config {
+    groups: ["queryRow", "query-simple-params"]
```
Comment (lahirusamith) [inline ballerina/tests/simple-params-query-test.bal:719]:
```suggestion
    groups: ["query", "query-simple-params"]
```

File: ballerina/tests/simple-params-query-test.bal [hunk 6]
```diff
@@ -673,6 +673,110 @@ function queryJsonParam3() returns record {}|error? {
...
+    if queryResult is error {
+        test:assertTrue(queryResult.message().endsWith("Unknown column 'invalid_column_name' in 'field list'."),
+                        "Incorrect error message");
+    } else {
+        test:assertFail("Expected error when querying with invalid column name.");
+    }
+}
+
+@test:Config {
+    groups: ["queryRow", "query-simple-params"]
```
Comment (lahirusamith) [inline ballerina/tests/simple-params-query-test.bal:736]:
```suggestion
    groups: ["query", "query-simple-params"]
```

File: ballerina/tests/simple-params-query-test.bal [hunk 7]
```diff
@@ -673,6 +673,110 @@ function queryJsonParam3() returns record {}|error? {
...
+function queryValue() returns error? {
+    Client dbClient = check new (host, user, password, simpleParamsDb, port);
+    string sqlQuery = "SELECT COUNT(*) FROM DataTable";
+    int count = check dbClient->queryRow(sqlQuery);
+    check dbClient.close();
+    test:assertEquals(count, 3);
+}
+
+@test:Config {
+    groups: ["queryRow", "query-simple-params"]
```
Comment (lahirusamith) [inline ballerina/tests/simple-params-query-test.bal:747]:
```suggestion
    groups: ["query", "query-simple-params"]
```

File: ballerina/tests/simple-params-query-test.bal [hunk 8]
```diff
@@ -673,6 +673,110 @@ function queryJsonParam3() returns record {}|error? {
...
+    if queryResult is error {
+        test:assertTrue(queryResult is sql:TypeMismatchError, "Incorrect error type");
+        test:assertEquals(queryResult.message(), "Expected type to be 'int' but found 'record{}'");
+    } else {
+        test:assertFail("Expected error when query result contains multiple columns.");
+    }
+}
+
+@test:Config {
+    groups: ["queryRow", "query-simple-params"]
```
Comment (lahirusamith) [inline ballerina/tests/simple-params-query-test.bal:764]:
```suggestion
    groups: ["query", "query-simple-params"]
```

File: ballerina/tests/simple-params-query-test.bal [hunk 9]
```diff
@@ -673,6 +673,110 @@ function queryJsonParam3() returns record {}|error? {
...
 
+@test:Config {
+    groups: ["queryRow","query-simple-params"]
+}
+function queryRecord() returns sql:Error? {
+    int rowId = 1;
+    sql:ParameterizedQuery sqlQuery = `SELECT * from DataTable WHERE row_id = ${rowId}`;
+    Client dbClient = check new (host, user, password, simpleParamsDb, port);
+    record {} queryResult = check dbClient->queryRow(sqlQuery);
+    validateDataTableResult(queryResult);
```
Comment (lahirusamith) [inline ballerina/tests/simple-params-query-test.bal:684]: You need to close the `dbClient` after using it.

File: ballerina/tests/simple-params-query-test.bal [hunk 10]
```diff
@@ -673,6 +673,110 @@ function queryJsonParam3() returns record {}|error? {
...
+    sql:ParameterizedQuery sqlQuery = `SELECT * from DataTable WHERE row_id = ${rowId}`;
+    Client dbClient = check new (host, user, password, simpleParamsDb, port);
+    record {}|sql:Error queryResult = dbClient->queryRow(sqlQuery);
+    if queryResult is sql:Error {
+        test:assertTrue(queryResult is sql:NoRowsError);
+        test:assertTrue(queryResult.message().endsWith("Query did not retrieve any rows."), "Incorrect error message");
+   } else {
+       test:assertFail("Expected no rows error with empty query result.");
+   }
+}
```
Comment (lahirusamith) [inline ballerina/tests/simple-params-query-test.bal:701]: same here

**Review by lahirusamith — COMMENTED** [review]
close `dbClient` after complete calling remote methods.

File: ballerina/tests/simple-params-query-test.bal (hunk 10 above)
Comment (kaneeldias) [reply ballerina/tests/simple-params-query-test.bal:703]: Fixed in 095d9d5

File: ballerina/tests/simple-params-query-test.bal (hunk 9 above)
Comment (kaneeldias) [reply ballerina/tests/simple-params-query-test.bal:685]: Fixed in 095d9d5

File: changelog.md (hunk 1 above)
Comment (kaneeldias) [reply changelog.md:18]: Fixed in 095d9d5

File: ballerina/tests/simple-params-query-test.bal (hunk 2 above)
Comment (kaneeldias) [reply ballerina/tests/simple-params-query-test.bal]: Fixed in 095d9d5

File: ballerina/tests/simple-params-query-test.bal (hunk 3 above)
Comment (kaneeldias) [reply ballerina/tests/simple-params-query-test.bal]: Fixed in 095d9d5

File: ballerina/tests/simple-params-query-test.bal (hunk 4 above)
Comment (kaneeldias) [reply ballerina/tests/simple-params-query-test.bal]: Fixed in 095d9d5

File: ballerina/tests/simple-params-query-test.bal (hunk 5 above)
Comment (kaneeldias) [reply ballerina/tests/simple-params-query-test.bal]: Fixed in 095d9d5

**Review by niveathika — APPROVED** [review]

**Review by daneshk — APPROVED** [review]

**Pull request merged**